### PR TITLE
indilib pkg, added support for aarm64, added snick as maintainer  

### PR DESCRIPTION
**Pull request by i-am-logger**
Hi @hjones2199 ,
I'm one of the developers for sky360.org and I want to add support for indilib QHY firmwares for raspberry pi 4. Was wondering if you can share some thoughts?

I created 3 issues with some dependencies to start.

https://github.com/NixOS/nixpkgs/issues/200135 'indi-full' support for aarch64
https://github.com/NixOS/nixpkgs/issues/200154 QHY CCD SDK
https://github.com/NixOS/nixpkgs/issues/200152 fxload does not support USB3 devices using the -fx3 flag

Best,

snick

### CODE REVIEW

**Comment by realsnick** [comment]
@hjones2199 any chance you can review this PR?

**Comment by realsnick** [comment]
@hjones2199 can you review this PR?

File: pkgs/development/libraries/science/astronomy/indilib/default.nix [hunk 1]
```diff
@@ -52,7 +56,7 @@ stdenv.mkDerivation rec {
     description = "Implementation of the INDI protocol for POSIX operating systems";
     changelog = "https://github.com/indilib/indi/releases/tag/v${version}";
     license = licenses.lgpl2Plus;
-    maintainers = with maintainers; [ hjones2199 ];
-    platforms = platforms.linux;
+    maintainers = with maintainers; [ hjones2199 snick ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/default.nix:60]:
```suggestion
    platforms = platforms.unix;
```

also systemd is not supported in darwin so this likely should stay like it is

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix [hunk 2]
```diff
@@ -31,8 +34,25 @@ stdenv.mkDerivation rec {
...
-    libjpeg libgphoto2 libraw libftdi1 libdc1394 ffmpeg fftw
+    indilib
+    libnova
+    curl
+    cfitsio
+    libusb1
+    systemd
+    zlib
+    boost
+    pkg-config
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix:45]: pkg-config goes into nativeBuildInputs

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix [hunk 3]
```diff
@@ -73,7 +93,7 @@ stdenv.mkDerivation rec {
     description = "Third party drivers for the INDI astronomical software suite";
     changelog = "https://github.com/indilib/indi-3rdparty/releases/tag/v${version}";
     license = licenses.lgpl2Plus;
-    maintainers = with maintainers; [ hjones2199 ];
-    platforms = platforms.linux;
+    maintainers = with maintainers; [ hjones2199 snick ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix:97]: systemd is not supported on darwin

File: pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix [hunk 4]
```diff
@@ -29,8 +31,24 @@ stdenv.mkDerivation rec {
...
 
   buildInputs = [
-    indilib libnova curl cfitsio libusb1 zlib boost gsl gpsd
-    libjpeg libgphoto2 libraw libftdi1 libdc1394 ffmpeg fftw
+    indilib
+    libnova
+    curl
+    cfitsio
+    libusb1
+    pkg-config
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix:39]: same here and for meta

File: pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix [hunk 5]
```diff
@@ -51,6 +69,11 @@ stdenv.mkDerivation rec {
     done
   '';
 
+  preFixup = ''
+    # fix missing libudev
+    ln -s ${lib.getLib systemd}/lib/libudev.so.1 $out/lib/libudev.so.1
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix:75]: This should probably be fixed with patchelf. Also this likely creates conflicts when combined in one environment.

**Review by SuperSandro2000 — COMMENTED** [review]
Please follow the [contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes) when naming your commits.

Also please remove any merge commit from the PR.

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix (hunk 2 above)
Comment (realsnick) [reply pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix:45]: can you please provide more information, what is NativeBuildInputs?

File: pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix (hunk 5 above)
Comment (realsnick) [reply pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix:75]: what is PatchIf? 

Thank you for all the help! much appreciated

File: pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/science/astronomy/indilib/indi-3rdparty.nix:45]: https://nixos.org/manual/nixpkgs/unstable/#variables-specifying-dependencies

File: pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/development/libraries/science/astronomy/indilib/indi-firmware.nix:75]: patchelf is a tool to manipulate .so files. You probably want something like https://github.com/NixOS/nixpkgs/blob/c48a958765d67d6f7e74ad8ab194440c3e8d1d08/pkgs/applications/editors/emacs/generic.nix#L239

**Comment by wegank** [comment]
Closed by #243071 for the title. If there are any other changes, please open a new PR, thanks.

**Pull request closed without merging**